### PR TITLE
nixos/*: Add types to the database module options

### PR DESCRIPTION
**Pull request by dasJ**
Add types to all database module options. See each commit for easier reviewing.

See also #76184

Some pings (sorry for bothering you):
- redis: @Mic92 
- virtuoso: nobody (is anyone even using the module?)
- firebird: nobody
- clickhouse: @Ma27 
- mongodb: @phile314 
- memcached: @dasJ (oh that's me)

### CODE REVIEW

**Comment by dasJ** [comment]
@GrahamcOfBorg test memcached mongodb clickhouse firebird virtuoso redis

File: nixos/modules/services/databases/firebird.nix [hunk 1]
```diff
@@ -64,20 +59,23 @@ in
 
       port = mkOption {
         default = "3050";
```
Comment (Ma27) [inline nixos/modules/services/databases/firebird.nix]: Shouldn't we use an int here if that option now represents a port?

File: nixos/modules/services/databases/redis.nix [hunk 2]
```diff
@@ -46,16 +46,12 @@ in
...
-        default = false;
-        description = ''
-          Whether to enable the Redis server. Note that the NixOS module for
-          Redis disables kernel support for Transparent Huge Pages (THP),
-          because this features causes major performance problems for Redis,
-          e.g. (https://redis.io/topics/latency).
-        '';
-      };
+      enable = mkEnableOption ''
+        Whether to enable the Redis server. Note that the NixOS module for
```
Comment (Ma27) [inline nixos/modules/services/databases/redis.nix]: `mkEnableOption`'s description is basically `Whether to enable <name>`, while `<name>` is the first argument passed to `mkEnableOption`. Thereforce I think that your text might seem rather confusing when looking at it from the manual. I'd suggest that you either adjust your text or that you do e.g. `(mkEnableOption "opt-name") // { description = ''A little more details''; }`.

File: nixos/modules/services/databases/memcached.nix [hunk 3]
```diff
@@ -16,45 +16,46 @@ in
 
   options = {
 
-    services.memcached = {
+    services.memcached = with types; {
```
Comment (Ma27) [inline nixos/modules/services/databases/memcached.nix]: (nitpick, probably irrelevant) I'm not sure if we actually benefit from adding even more `with ...;` expressions. While I think that those are pretty handy and convenient, I know that especially beginners will have a hard time understanding where a variable is *actually* coming from if it was passed through several expressions using `with ...;`.

**Review by Ma27 — APPROVED** [review]
Thanks a lot for taking care of this, unfortunately I didn't manage to leave some useful feedback earlier - sorry! I left three (rather trivial) comments containing stuff I noticed, feel free to comment/resolve (if those are invalid :)).

**Review by phile314 — APPROVED** [review]
Only looked at the mongodb changes, but those look good to me.

**Comment by Mic92** [comment]
I rebased it here: https://github.com/NixOS/nixpkgs/pull/91813

**Pull request closed without merging**